autoload/sj/r.vim

```diff
@@ -0,0 +1,190 @@
+" Only real syntax that's interesting is cParen and cConditional
+let s:skip = sj#SkipSyntax(['rComment'])
+
+" function! sj#r#SplitFuncall()
+"
+" Split the R function call if the cursor lies within the arguments of a
+" function call
+"
+function! sj#r#SplitFuncall()
+  if !s:IsValidSelection("va(")
+    return 0
+  endif
+
+  call sj#PushCursor()
+  let items = s:ParseJsonFromMotion("va(\<esc>vi(")
+  let items = map(items, {k, v -> v . (k+1 < len(items) ? "," : "")})
+
+  if g:r_indent_align_args && len(items)
+    let items[0]  = "(" . items[0]
+    let items[-1] = items[-1] . ")"
+    let lines = items
+  else
+    let lines = ["("] + items + [")"]
+  endif
+
+  call sj#PopCursor()
+  call s:ReplaceMotionPreserveCursor('va(', lines)
+
+  return 1
+endfunction
+
+" function! sj#r#JoinFuncall()
+"
+" Join an R function call if the cursor lies within the arguments of a
+" function call
+"
+function! sj#r#JoinFuncall()
+  if !s:IsValidSelection("va(")
+    return 0
+  endif
+
+  call sj#PushCursor()
+  let items = s:ParseJsonFromMotion("va(\<esc>vi(")
+
+  " clean up unwanted spaces around parens which can occur during nested joins
+  let text = join(items, ", ")
+  let text = substitute(text, '(\s', '(', 'g')
+  let text = substitute(text, '\s)', ')', 'g')
+
+  call sj#PopCursor()
+  call s:ReplaceMotionPreserveCursor("va(", ["(" . text . ")"])
+
+  return 1
+endfunction
+
+" function! s:Triml(text [, mask])
+"
+" A shorthand for trim with setting dir=1
+"
+function! s:Triml(text, ...)
+  let inlen = len(a:text)
+  let text = a:0 > 0
+        \ ? trim(a:text, a:1, 1)
+        \ : substitute(a:text, '^\s*', '', 'g')
+  return [text, inlen - len(text)]
+endfunction
```

Comment:
Additionally, the build doesn't like the usage of `trim(` (I think), since it's been added in `8.0.1630` and I guess the latest Vim in debian stretch is `8.0.1453` :). And the mask is not even used, so I feel like there's no point in relying on it.

Reply:
Thanks! I must have missed `Ltrim` when poking around for something similar. I'll make that change.

Reply:
I removed my version and use `sj#Ltrim`. The number of leading whitespace characters was only needed in onplace, so I just calculate those where they're needed now.